Assets/Scripts/Game/Entities/EnemyEntity.cs

```diff
@@ -244,7 +244,7 @@ public override void ClearConstantEffects()
         /// <summary>
         /// Sets enemy career and prepares entity settings.
         /// </summary>
-        public void SetEnemyCareer(MobileEnemy mobileEnemy, EntityTypes entityType)
+        public virtual void SetEnemyCareer(MobileEnemy mobileEnemy, EntityTypes entityType)
         {
```

Comment:
Note that this `virtual` isn't strictly necessary for mods. All the fields `SetEnemyCareer` modifies are public, so mods can register to `OnLootSpawned` and change back the fields from there.

But sure, it's cleaner if your custom entity type can just set them right to begin with

Reply:
I use `SetEnemyCareer` to do a couple of things besides set field values. I could try to manually erase all generated items from every enemy and respawn them myself, for example, but like you said it'll be a lot cleaner if I can change this directly.